### ISSUE TITLE
Discussion: AudioServlet and wav files

### ISSUE THREAD

**Issue by t2000** [openhab/openhab-core#1350]
I have just debugged why the PicoTTS service is not working with my Squeezebox installation.

I am running the Picoreplayer project with Logitech Media Server and squeezelite instances.

PicoTTS produces ".wav" files and openHAB serves them on its webserver. Browsers like Chromium are able to play it from there. 

The squeezelite instance however, got the URL in its playlist correctly, but refuses to play it.

First I thought that my LMS or squeezelite has problems with wave files in a certain format, like earlier versions of these programs did, but that is not the case for me. I could add the file downloaded from openhab on the harddisk of the picoreplayer and play it from there without issues.

Then I setup an apache webserver on my development machine and served the wav file from there. Result was that the file could be played by squeezelite without issues!

Wireshark helped me to find the difference between apache and the openHAB audio servlet. It is right here:

https://github.com/openhab/openhab-core/blob/e63db5f5c56bdb370707a395b78ee4d7b215a9d3/bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/internal/AudioServlet.java#L100

Apache uses the type `audio/x-wav` instead.

For testing purposes I have edited the openHAB servlet to serve the wav files as `audio/x-wav` too. This is working perfectly fine. Squeezelite plays them without any issues!

by searching the internet I found that this isn't a new issue:
https://github.com/jshttp/mime-db/issues/22

However, their recommendation `audio/vnd.wave` from http://tools.ietf.org/html/rfc2361 doesn't work for the squeezebox case.

Long story short: This leaves me to the question whether we could simply replace the `audio/wav` by `audio/x-wav` or if we have to enhance the code so we can somehow dynamically decide which mimetype to use.

Any suggestions?

AFAIK picoTTS is the **only** TTS service supported by openHAB that does NOT need a cloud connection, that is why I really want to use it!

**Comment by kaikreuzer** [comment]
> whether we could simply replace the audio/wav by audio/x-wav

Hm, in the linked issue it is said that `audio/wav` is probably the best choice and this is what we are doing right now. It is hence rather a buggy situation in the squeezebox software.

As you are saying that you're using LMS, wouldn't the best place to raise the issue be https://github.com/Logitech/slimserver?

**Comment by openhab-5iver** [comment]
> AFAIK picoTTS is the only TTS service supported by openHAB that does NOT need a cloud connection, that is why I really want to use it!

You can also use [flite](http://www.festvox.org/flite/), which is **much** more robust than pico2wav. That is not the best link, but I will put a tutorial together in the forum for how I use it, including adjusting the voice, conversion to mp3, and appending an alert prefix sound, so you know there is an alert playing and not random people talking in the house!
